### PR TITLE
Obsolete Workflow command "set-output"

### PR DESCRIPTION
**Pull request by bashbang**
## Description
The Github Action command set-output is being obsoleted in favour of using env $GITHUB_OUTPUT. See details here: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
This PR migrates to the new method.
Some of the workflows what run windows environments needed to the shell explicitly set to bash, so I conformed them all for consistency.
Note: There was one action that had the command commented out. This was also updated to the new method in case it's uncommented in the future.

## How has this been tested?
- A run of the pr-tests-* all passed except pr-tests-stack which was unable to run due to unavailability of om-ci-16vcpu-ubuntu2204. However the same method was implemented in all workflows so expect that the other workflows should work as expected.

## Checklist
- [X] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [X] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [X] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [X] My changes are covered by tests

### CODE REVIEW

File: .github/workflows/pr-tests-stack.yml [hunk 1]
```diff
@@ -261,8 +264,9 @@ jobs:
       - name: Get pip cache dir
         if: steps.changes.outputs.stack == 'true'
         id: pip-cache
+        shell: bash
         run: |
-          echo "::set-output name=dir::$(pip cache dir)"
+          echo "dir=$(pip cache dir)" >> %GITHUB_OUTPUT
```
Comment (kiendang) [inline .github/workflows/pr-tests-stack.yml]:
```suggestion
          echo "dir=$(pip cache dir)" >> $GITHUB_OUTPUT
```

Comment (kiendang) [reply .github/workflows/pr-tests-stack.yml]: Great work. Thank you! Just 1 small fix.

**Comment by rasswanth-s** [comment]
@bitsofsteve  PTAL

File: .github/workflows/pr-tests-stack.yml (hunk 1 above)
Comment (bashbang) [reply .github/workflows/pr-tests-stack.yml]: Good catch!

**Review by kiendang — APPROVED** [review]

**Review by bitsofsteve — APPROVED** [review]
@bashbang LGTM 👍🏾, Great work spotting these! I'll merge this now.

**Pull request merged**